Debug App/Sources/Utilities/MetadataParser.swift

```diff
@@ -0,0 +1,51 @@
+//
+//  MetadataParser.swift
+//  Debug App
+//
+//  Created by Niall Quinn on 11/03/24.
+//  Copyright © 2024 Primer API Ltd. All rights reserved.
+//
+
+import Foundation
+
+
+struct MetadataParser {
+
+    func parse(_ metadata: String?) -> [String: Any] {
+        guard let metadata = metadata else { return [:] }
+
+        if let jsonData = metadata.data(using: .utf8),
+           let jsonObject = try? JSONSerialization.jsonObject(with: jsonData, options: []) as? [String: Any] {
+            return jsonObject
+        }
+
+        let keyValuePairs = metadata
+            .split(separator: "\n")
+            .map { $0.split(separator: "=").map { String($0).trimmingCharacters(in: .whitespaces) } }
+            .filter { $0.count > 1 }
+            .map { (keyValue: [String]) -> (String, Any) in
+                let (key, value) = (keyValue[0], keyValue[1])
+                let parsedValue = parseValue(value)
+                return (key, parsedValue)
+            }
+
+        return Dictionary(uniqueKeysWithValues: keyValuePairs)
+    }
+
+    private func parseValue(_ value: String) -> Any {
+        // Boolean
+        if value.lowercased() == "true" { return true }
+        if value.lowercased() == "false" { return false }
+
+        if let number = Double(value) { return number }
+        if let int = Int(value) { return int }
+        
+        // String
+        if value.hasPrefix("\"") && value.hasSuffix("\"") {
+            return String(value.dropFirst().dropLast())
+        }
+
+        // the rest is returned as is
+        return value
+    }
+}
```

Comment:
I asked GPT if it can optimise this parsing a bit, and there were a few points, I don't want to play smart by copy/pasta these answers, here is the link so you can read it yourself. https://chat.openai.com/share/d46123f6-2304-4a15-9f7d-2bb7fe58dabc

Reply:
Ill take a look, I started with .reduce, but the compiler complained (expression is too complex to parse).
Will revisit

Reply:
1: The `parse` section of the ChatGPT gives this error (as I saw previously) `The compiler is unable to type-check this expression in reasonable time; try breaking up the expression into distinct sub-expressions`. Thanks Xcode.

2: The parsing ChatGPT suggested as a whole wont work for us as case matters, but I will take one suggestion
`Number Parsing Optimization: Check for integer values before floating-point values because Int parsing is generally faster than Double, and integers are more common in many contexts.`

Thanks!